tests/miral/test_input_configuration.cpp

```diff
@@ -0,0 +1,690 @@
+    {
+        mouse_config.acceleration(value);
+        EXPECT_THAT(mouse_config.acceleration(), Eq(value));
+    }
+}
+
+TEST_F(TestMouseInputConfiguration, mouse_acceleration_bias_returns_expected_value)
+{
+    auto const value = -1;
+    mouse_config.acceleration_bias(value);
+    EXPECT_THAT(mouse_config.acceleration_bias(), Eq(value));
+}
+
+TEST_F(TestMouseInputConfiguration, mouse_handedness_returns_expected_value)
+{
+    auto const values = {mir_pointer_handedness_left, mir_pointer_handedness_right};
+    for(auto const value: values)
+    {
+        mouse_config.handedness(value);
+        EXPECT_THAT(mouse_config.handedness(), Eq(value));
+    }
+}
+
+TEST_F(TestMouseInputConfiguration, mouse_hscroll_speed_returns_expected_value)
+{
+    auto const value = -5;
+    mouse_config.hscroll_speed(value);
+    EXPECT_THAT(mouse_config.hscroll_speed(), Eq(value));
+}
+
+TEST_F(TestMouseInputConfiguration, mouse_vscroll_speed_returns_expected_value)
+{
+    auto const value = 5;
+    mouse_config.vscroll_speed(value);
+    EXPECT_THAT(mouse_config.vscroll_speed(), Eq(value));
+}
+
+struct TestMouseMergeOneProperty : public TestInputConfiguration, testing::WithParamInterface<MouseProperty>
+{
+    Mouse target;
+};
+
+// Make sure that when merging, each data member is assigned to the correct
+// corresponding one.
+//
+// If each individual data member is merged correctly, then all combinations
+// will work.
+TEST_P(TestMouseMergeOneProperty, mouse_merge_from_partial_set_changes_only_set_values)
+{
+    auto const property = GetParam();
+
+    auto const expected = get_mouse_config_with_properties(std::array{property});
+    auto const to_merge = get_mouse_config_with_properties(std::array{property});
+
+    target.merge(to_merge);
+
+    mouse_expect_equal(target, expected);
+}
+
+INSTANTIATE_TEST_SUITE_P(TestInputConfiguration, TestMouseMergeOneProperty, ::testing::ValuesIn(all_mouse_props));
+
+struct TestMouseMergeOnePropertyWithoutOverwrite :
+    public TestInputConfiguration,
+    testing::WithParamInterface<MouseProperty>
+{
+};
+
+// Make sure tht merging does not alter other data members with the same typ
+TEST_P(
+    TestMouseMergeOnePropertyWithoutOverwrite,
+    mouse_merge_source_property_does_not_override_different_properties_with_the_same_type)
+{
+    auto const property = GetParam();
+
+    auto const expected = get_mouse_config_with_properties(all_mouse_props);
+    auto const modified = get_mouse_config_with_properties(std::array{property});
+    auto target = get_mouse_config_with_all_properties_except(property);
+
+    target.merge(modified);
+
+    mouse_expect_equal(target, expected);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    TestInputConfiguration, TestMouseMergeOnePropertyWithoutOverwrite, ::testing::ValuesIn(all_mouse_props));
+
+// Make sure tht merging with properties already set results in no change.
+TEST_F(TestMouseMergeOnePropertyWithoutOverwrite, mouse_merge_with_properties_already_set_does_not_change_target)
```

Comment:
The test fixture name is repeated in the test name
```suggestion
TEST_F(TestMouseMergeOnePropertyWithoutOverwrite, does_not_change_target_properties_that_are_already_set)
```